std/heap.zig

```diff
@@ -360,3 +382,32 @@ fn testAllocator(allocator: &mem.Allocator) !void {
 
     allocator.free(slice);
 }
+
+fn testAllocatorLargeAlignment(allocator: &mem.Allocator) mem.Allocator.Error!void {
+    //Maybe a platform's page_size is actually the same as or 
+    //  very near usize?
+    if(os.page_size << 2 > @maxValue(usize)) return;
```

Comment:
Isn't this logically always false if `os.page_size == @maxValue(usize)`?  As well, this check is the kind of hypothetical I wouldn't worry about until it becomes an actual problem.

Reply:
Since everything involved is a literal, I don't believe there is any overflow to worry about. However, @maxValue(usize) really should be (2 << usize.bit_count). Of course this all assumes po2 page size, but I don't think that's worth worrying about until LLVM supports a ternary native architecture.